### ISSUE TITLE
[2.0.x] Registry isset/empty don't work

### ISSUE THREAD

**Issue by wojewsky** [phalcon/cphalcon#10267]
I've a problem with registry and isset.
This is my definition:

```php
$this->_di->set('registry', function() {
  //Registry erzeugen
  return new \Phalcon\Registry();
});
...
$this->di['registry']->test = 'abc';
$test = (isset($this->di['registry']->test)) ? true : false;
die(var_dump($test));
```

I've got: false

**Comment by dreamsxin** [comment]
Please use `setShared`

**Comment by wojewsky** [comment]
I've the same problem with:
$this->_di->setShared('registry', function() {
  return new \Phalcon\Registry();
});

**Comment by dreamsxin** [comment]
I test it in lasted version 2.0.x, it's right.
```php
$di = new Phalcon\DI();

$di->set('registry', function() {
	return new \Phalcon\Registry();
});

$di['registry']->test = 'abc';
$test = (isset($di['registry']->test)) ? true : false;
var_dump($test);
``

**Comment by KorsaR-ZN** [comment]
It's works in branch `2.0.x`
In `master` or `2.0.0` - don't work, see #10117

```
echo 'Phalcon v.' . \Phalcon\Version::get();
echo PHP_EOL;

$di = new Phalcon\DI();
$di->set('registry', function() {
    return new \Phalcon\Registry();
});

$di['registry']->test = 'abc';

var_dump(isset($di['registry']->test));
var_dump(isset($di['registry']->test2));

echo PHP_EOL;
echo '= false' . PHP_EOL;
$di['registry']->test = false;
var_dump(isset($di['registry']->test));

echo PHP_EOL;
echo '= null' . PHP_EOL;
$di['registry']->test = null;
var_dump(isset($di['registry']->test));


echo PHP_EOL;
echo 'unset()' . PHP_EOL;
unset($di['registry']->test);
var_dump(isset($di['registry']->test));


echo PHP_EOL;
var_dump(empty($di['registry']->test));

$di['registry']->test = 'abc';
var_dump(empty($di['registry']->test));
```

**Outputs**
```

Phalcon v.2.0.1
bool(true)
bool(false)

= false
bool(true)

= null
bool(true)

unset()
bool(false)

bool(true)
bool(false)
```

**Comment by andresgutierrez** [comment]
Fixed in 2.0.1
